src/feature/flow_chart_panel/views/APIKeyModal.tsx

```diff
@@ -1,76 +1,147 @@
-import { ChangeEvent, memo } from "react";
+import FamilyHistoryIconSvg from "@src/assets/family_history_icon";
+import { ChangeEvent, memo, useState } from "react";
 import { Modal, createStyles, Button, Input } from "@mantine/core";
+import { Notifications, notifications } from "@mantine/notifications";
 import { useFlowChartState } from "@src/hooks/useFlowChartState";
-import { sendApiKeyToDjango } from "@src/services/FlowChartServices";
+import { IconCheck } from "@tabler/icons-react";
 
 interface APIKeyModelProps {
   isOpen: boolean;
   onClose: () => void;
 }
 
+const BACKEND_HOST = process.env.VITE_SOCKET_HOST || "localhost";
+const BACKEND_PORT = +process.env.VITE_BACKEND_PORT! || 8000;
+const API_URL = "http://" + BACKEND_HOST + ":" + BACKEND_PORT;
+
 const useStyles = createStyles((theme) => ({
   container: {
-    display: "flex",
+    display: "relative",
     justifyContent: "center",
     alignItems: "center",
     gap: 43,
-    height: "100%",
+    height: 202,
     width: "100%",
-    padding: 24,
-    backgroundColor: theme.colors.modal[0],
+    backgroundColor: theme.colors.modal[1],
+    borderRadius: 17,
   },
   title: {
+    display: "flex",
+    gap: 5,
     fontSize: 24,
     fontWeight: "bold",
     fontFamily: "Inter",
-    marginBottom: 10,
+    marginTop: "10%",
+    marginLeft: "22%",
   },
-  submitButton: {
+  submitButtonLine: {
     display: "flex",
+    justifyContent: "center",
+    alignItems: "center",
     gap: 5,
+    color: theme.colors.accent1[0],
+  },
+  submitBtn: {
+    backgroundColor:
+      theme.colorScheme === "dark" ? theme.colors.accent1[0] : "none",
+    color: theme.colorScheme === "dark" ? theme.colors.modal[1] : "none",
+  },
+  closeBtn: {
+    position: "absolute",
+    top: 10,
+    right: 15,
+  },
+  inputBox: {
+    input: {
+      width: 250,
+      backgroundColor: theme.colors.modal[0],
+    },
   },
 }));
-
 const APIKeyModal = ({ isOpen, onClose }: APIKeyModelProps) => {
   const { classes } = useStyles();
   const { apiKey, setApiKey } = useFlowChartState();
+  const [apiResponse, setApiResponse] = useState({
+    success: false,
+    data: null,
+  });
 
   const handleApiKeyChange = (e: ChangeEvent<HTMLInputElement>) => {
     setApiKey(e.target.value);
   };
-
   const handleClose = () => {
     setApiKey("");
     onClose();
   };
-
+  const sendApiKeyToDjango = async (apiKey: string) => {
```

Comment:
Any reason for not using `sendApiKeyToDjango` function from `@src/services/FlowChartServices` ? if it is for notification then we can modify the function to return a success or failure message and show from here.